pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -515,13 +539,22 @@ func (s *Server) createSelfSignedCACertificateOptions(fileBundle *ca.SigningCAFi
 // createIstioRA initializes the Istio RA signing functionality.
 // the caOptions defines the external provider
 // ca cert can come from three sources, order matters:
-// 1. Define ca cert via kubernetes secret and mount the secret through `external-ca-cert` volume
-// 2. Use kubernetes ca cert `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt` if signer is
-//
-//	kubernetes built-in `kubernetes.io/legacy-unknown" signer
-//
-// 3. Extract from the cert-chain signed by other CSR signer.
+// 1. Define ca cert via kubernetes secret and mount the secret through `external-ca-cert` volume mounted as ./etc/external-ca-cert/root-cert.pem
+// 2. Extract from the cert-chain signed by other CSR signer.
```

Comment:
Could we clarify that this is obtained from the meshConfig? These options don't clearly align with the logs below. Could we make the comments more consistent for readability? Thanks!

Reply:
Regardless of whether or not we load the ca cert via the external-ca-cert volume, do we get certs from the meshconfig? If that is the case can we clarify that with a comment?

Reply:
We actually don't get roots from meshconfig for the istio-ca case - or I didn't find them. I thought so (3 ways) - but it looks like meshconfig secrets are only used for the RA (K8S signing), and even there - if they are missing it falls back.

We should keep iterating on the comments and the logs until they reflect reality (and add TODOs to reflect the intent ) - but no need to comment/document all the code in a single PR :-).

Reply:
For the record: I am not a big fan of the meshconfig for root certs. 

Standardizing on one way - external-ca-certs - seems best option. The file is likely to be generated and operated by other tools ( like CertManager trust manager, or some CI/CD) - which may be good at generating a neutral config map or secret, but we can't expect them to patch MeshConfig or be aware of Istio.